idseq_pipeline/commands/accessionid2seq_functions.py

```diff
@@ -353,55 +370,59 @@ def get_sequence_by_accession_id_ntf(accession_id, nt_loc_dict, ntf):
-        accession_file = 'accession-%s' % accession_id
-        NUM_RETRIES = 3
-        for attempt in range(NUM_RETRIES):
+    accession_file = 'accession-%s' % accession_id
+    NUM_RETRIES = 3
+    pipe_file = ''
+    for attempt in range(NUM_RETRIES):
+        try:
+            pipe_file = 'pipe-{attempt}-accession-{accession_id}'.format(
+                attempt=attempt, accession_id=accession_id)
+            os.mkfifo(pipe_file)
+            get_range = "aws s3api get-object --range bytes=%d-%d --bucket %s --key %s %s" % (
+                range_start, range_start + name_length + seq_len - 1,
+                nt_bucket, nt_key, pipe_file)
+            get_range_proc = subprocess.Popen(
+                get_range, shell=True, stdout=DEVNULL)
+
+            cmd = "cat {pipe_file} | tee >(tail -n+2 | tr -d '\n' > {accession_file}) | head -1".format(
+                pipe_file=pipe_file, accession_file=accession_file)
+            seq_name = subprocess.check_output(
+                cmd, executable='/bin/bash', shell=True).split(" ", 1)[1]
+            exitcode = get_range_proc.wait()
+            msg = "Error in get_sequence_by_accession_id_s3."
+            assert exitcode == 0, msg
+            seq_len = os.stat(accession_file).st_size
+            break
+        except:
+            if attempt + 1 < NUM_RETRIES:
+                time.sleep(1.0 * (4**attempt))
+            else:
+                print(
+                    "All retries failed for get_sequence_by_accession_id_s3.")
+                raise
+        finally:
             try:
-                pipe_file = 'pipe-{attempt}-accession-{accession_id}'.format(
-                    attempt=attempt, accession_id=accession_id)
-                os.mkfifo(pipe_file)
-                get_range = "aws s3api get-object --range bytes=%d-%d --bucket %s --key %s %s" % (
-                    range_start, range_start + name_length + seq_len - 1,
-                    nt_bucket, nt_key, pipe_file)
-                get_range_proc = subprocess.Popen(
-                    get_range, shell=True, stdout=DEVNULL)
-                seq_name = subprocess.check_output(
-                    "cat {pipe_file} |tee >(tail -n+2 |tr -d '\n' > {accession_file}) |head -1 ".
-                    format(pipe_file=pipe_file, accession_file=accession_file),
-                    executable='/bin/bash',
-                    shell=True).split(" ", 1)[1]
-                exitcode = get_range_proc.wait()
-                msg = "Error in get_sequence_by_accession_id_s3."
-                assert exitcode == 0, msg
-                seq_len = os.stat(accession_file).st_size
-                break
-            except:
-                if attempt + 1 < NUM_RETRIES:
-                    time.sleep(1.0 * (4**attempt))
-                else:
-                    print("All retries failed for get_sequence_by_accession_id_s3.")
-                    raise
-            finally:
-                try:
+                if pipe_file != '':
```

Comment:
is the goal of this new check to get rid of try/except here?

Reply:
Oh it was actually because I got a linter warning that pipe_file could be undeclared so I instantiated pipe_file = '' above the block and added this line. But pipe_file is the first line in the 'try' so I took this out again.